### PR TITLE
refactor: PackageFile types

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Improves typing by adding type `PackageFile` which is `PackageFileContent` with filename.

## Context

Better types

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/onboarding/pr/index.ts [hunk 1]
```diff
@@ -86,7 +86,7 @@ If you need any further assistance then you can also [request help here](${
     for (const [manager, managerFiles] of Object.entries(packageFiles)) {
       files = files.concat(
         // TODO: types (#7154)
```
Comment (RahulGautamSingh) [inline lib/workers/repository/onboarding/pr/index.ts:88]:
```suggestion
```

File: lib/workers/repository/package-files.ts [hunk 2]
```diff
@@ -106,7 +106,7 @@ export class PackageFiles {
         deps += `<details><summary>${manager}</summary>\n<blockquote>\n\n`;
         for (const packageFile of packageFiles[manager]) {
           // TODO: types (#7154)
```
Comment (RahulGautamSingh) [inline lib/workers/repository/package-files.ts:108]:
```suggestion
```

**Review by RahulGautamSingh — CHANGES_REQUESTED** [review]
Some Todos not needed anymore

**Comment by rarkins** [comment]
Blocked by #20502

File: lib/modules/manager/maven/extract.ts [hunk 3]
```diff
@@ -249,15 +245,15 @@ function resolveParentFile(packageFile: string, parentPath: string): string {
...
   mavenProps?: Record<string, any>;
   parent?: string;
 }
 
 export function extractPackage(
   rawContent: string,
-  packageFile: string | null = null
-): PackageFileContent<Record<string, any>> | null {
+  packageFile: string
+): PackageFile<Record<string, any>> | null {
```
Comment (viceice) [inline lib/modules/manager/maven/extract.ts:256]:
```suggestion
): PackageFile | null {
```
remove default type

File: lib/modules/manager/maven/extract.ts [hunk 4]
```diff
@@ -462,7 +456,7 @@ export function resolveParents(
 
 function cleanResult(
   packageFiles: MavenInterimPackageFile[]
-): PackageFileContent<Record<string, any>>[] {
+): PackageFile<Record<string, any>>[] {
```
Comment (viceice) [inline lib/modules/manager/maven/extract.ts:459]:
```suggestion
): PackageFile[] {
```

**Review by viceice — COMMENTED** [review]
more later

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/modules/manager/maven/extract.ts [hunk 5]
```diff
@@ -249,15 +245,15 @@ function resolveParentFile(packageFile: string, parentPath: string): string {
...
   mavenProps?: Record<string, any>;
   parent?: string;
 }
 
 export function extractPackage(
   rawContent: string,
-  packageFile: string | null = null
-): PackageFileContent<Record<string, any>> | null {
+  packageFile: string
+): PackageFile | null {
```
Comment (viceice) [inline lib/modules/manager/maven/extract.ts:256]: needs prettifying

**Review by viceice — COMMENTED** [review]
otherwise LGTM

**Review by viceice — DISMISSED** [review]

**Review by viceice — APPROVED** [review]

**Review by secustor — APPROVED** [review]

**Pull request merged**